pyanaconda/modules/services/installation.py

```diff
@@ -123,3 +126,77 @@ def run(self):
         for service_name in self._enabled_services:
             log.debug("Enabling service: %s.", service_name)
             util.enable_service(service_name, root=self._sysroot)
+
+
+class ConfigureSystemdDefaultTargetTask(Task):
+    """Installation task for configuring systemd default target.
+
+    Set the correct systemd default target for the target system.
+
+    We support setting either the text only "multi-user"
+    target or the graphical target called "graphical".
+    """
+
+    SUPPORTED_SYSTEMD_TARGETS = [TEXT_ONLY_TARGET, GRAPHICAL_TARGET]
+
+    def __init__(self, sysroot, default_target=None):
+        """Create a new systemd target configuration task.
+
+        :param str sysroot: a path to the root of the target system
+        :param default_target: systemd default target to be set
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        # default to the text only target
+        if not default_target:
+            default_target = TEXT_ONLY_TARGET
+        self._default_target = default_target
+
+    @property
+    def name(self):
+        return "Configure systemd default target"
+
+    def run(self):
+        if self._default_target not in self.SUPPORTED_SYSTEMD_TARGETS:
+            msg = "Unsupported systemd default target: {} Specify one of: {}".format(
+                self._default_target, self.SUPPORTED_SYSTEMD_TARGETS
+            )
+            raise SystemdDefaultTargetConfigurationError(msg)
+        else:
+            log.debug("Setting systemd default target to: %s", self._default_target)
+
+        if not os.path.isdir(os.path.join(self._sysroot, 'etc/systemd/system')):
+            msg = "There is no /etc/systemd/system directory, cannot set systemd default target!"
+            raise SystemdDefaultTargetConfigurationError(msg)
+
+        default_target_path = os.path.join(self._sysroot, 'etc/systemd/system/default.target')
+        # unlike an links already in place
```

Comment:
I don't understand the comment.

Reply:
should have been "unlink any links already in place", fixed